File: ruby/codeabbey/010_bmi.rb
#!/usr/bin/env ruby
# encoding: utf-8

x = ARGF.file.read.split "\n"
x.shift
x.map! do |e|
  e = e.split(' ')
  bmi = e[0].to_f / (e[1].to_f**2)
  if bmi < 18.5
    'under'
  elsif bmi < 25
    'normal'
  elsif bmi < 30
    'over'
  else
    'obese'
  end
end
puts x.join ' '

File: codeabbey/010-bmi.js
#!/usr/bin/env node

var input, output;

input = require('fs')
        .readFileSync('./input.txt', 'utf8')
        .split('\n')
        .slice(1, -1)
        .map(function(nums) {
          return nums.split(' ')
            .map(function(num) {
              return parseFloat(num, 10);
            });
        });

output = input.map(function(nums) {
  var bmi = nums[0] / Math.pow(nums[1], 2);

  if      (bmi < 18.5) { return 'under';  }
  else if (bmi < 25)   { return 'normal'; }
  else if (bmi < 30)   { return 'over';   }
  else                 { return 'obese';  }

}).join(' ');

console.log(output);
